### DISCUSSION TITLE
Add a new classes into Yolov8 trained model

### DISCUSSION THREAD

**Discussion by Ace-Solomon** [ultralytics/ultralytics#7213] [Q&A]
Hello guys , I have a model trained to detect the person. 
now I wanted to add another class into the model to detect the person face.
what should I wrote on the yaml files ?

**Comment by pderrenger** [comment]
@Ace-Solomon hello! To add a new class to your YOLOv8 model that has already been trained to detect persons, you'll need to update your dataset and configuration files accordingly. Here's a concise guide on what to do:

1. **Update Dataset**: Add images containing the new class (person faces) to your dataset. Make sure to annotate these images with bounding boxes for the new class.

2. **Modify YAML Files**:
   - **Data YAML**: In your `data.yaml` file, you'll need to increment the number of classes by one and add the new class label to the `names` list. For example:
     ```yaml
     nc: 2  # number of classes
     names: ['person', 'face']  # class names
     ```
   - **Model YAML**: If you're using a custom model configuration (`*.yaml`), ensure the `nc` (number of classes) in the last layer matches the new number of classes in your `data.yaml`.

3. **Retrain the Model**: With the updated dataset and configuration, you'll need to retrain your model. You can either train from scratch or use transfer learning from your existing weights. For transfer learning, load your model with the updated `nc` and train on the new dataset.

Here's an example of how you might start training with transfer learning:
```python
from ultralytics import YOLO

# Load your existing model with updated number of classes
model = YOLO('path/to/your/model.yaml', pretrained='path/to/your/weights.pt')

# Train on the new dataset with the additional class
results = model.train(data='path/to/updated/data.yaml', epochs=50)
```

Remember to adjust paths and parameters according to your specific setup. Good luck with adding the new class to your model! 😊🚀
